### PR TITLE
Qthreads backend

### PR DESCRIPTION
**Pull request by stelleg**
Qthreads backend. Will submit the necessary tapir-clang PR shortly.

### CODE REVIEW

File: include/llvm/Transforms/Utils/TapirUtils.h [hunk 1]
```diff
@@ -94,8 +94,7 @@ class LoopSpawningHints {
     case LoopSpawningHints::ST_DAC:
       return "Use divide-and-conquer";
     case LoopSpawningHints::ST_END:
-    default:
-      return "Unknown";
+      return "ST_END";
```
Comment (wsmoses) [inline include/llvm/Transforms/Utils/TapirUtils.h]: why change this?

Comment (wsmoses) [reply include/llvm/Transforms/Utils/TapirUtils.h]: (perhaps keep to return unknown?)

File: test/Transforms/Tapir/qthreads-abi-calls.ll [hunk 2]
```diff
@@ -0,0 +1,68 @@
+; Simple test to verify that Tapir lowering to the Cilk Plus target
+; inserts the correct Cilk ABI calls.
+;
+; RUN: opt < %s -tapir2target -tapir-target=qthreads -debug-abi-calls -S | FileCheck %s
+
```
Comment (wsmoses) [inline test/Transforms/Tapir/qthreads-abi-calls.ll]: no need for debug-abi-calls

**Review by wsmoses — CHANGES_REQUESTED** [review]

**Comment by stelleg** [comment]
@wsmoses Comments should be addressed

**Review by wsmoses — DISMISSED** [review]

**Review by wsmoses — APPROVED** [review]

**Pull request merged**